optuna/study/study.py

```diff
@@ -392,9 +398,6 @@ def objective(trial):
                 when ``n_trials`` is :obj:`None``, ``timeout`` not is :obj:`None`, and
                 ``n_jobs`` :math:`\\ne 1`.
 
-        Raises:
-            RuntimeError:
-                If nested invocation of this method occurs.
```

Comment:
I think this error is non-obvious, so we should leave this as it is.

Reply:
Do you think `note` is insufficient for noticing users? I agree with you that it is not obvious but I'm not sure if we have to make `RuntimeError` the specification.

https://github.com/optuna/optuna/pull/3429/files#diff-0c6709b7aa4878157a8d4f936a0a9c523a2f9ced2866610bf3ed39083abe5952R325-R326

Reply:
We discussed this privately and decided to revert here.

Reply:
[cdaa735](https://github.com/optuna/optuna/pull/3429/commits/cdaa7358249f92d5976f9cf9596e56aadb591840)